### DISCUSSION TITLE
How to perform an offline install of yarn berry?

### DISCUSSION THREAD

**Discussion by Paper-Folding** [yarnpkg/berry#6360] [Q&A] [answered]
My company don't allow my machine to access internet (nor use a proxy server). The only thing I can do is copy files from outside to inside dev machine. While `yarn set version berry` will download files for installation, how can I install yarn berry offline? Is there any guideline? Thank you.

**Comment by arcanis** [comment]
You can copy the [`yarn.js` file](https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-cli/bin/yarn.js).

**Reply by Paper-Folding** [reply]
Is this that simple? So I can execute any yarn CLI command like `node yarn.js install` ?

**Reply by Paper-Folding** [reply]
Yes, `node yarn.js` work like a charm, thank you!

**Comment by clemyan** [comment, marked as answer]
If you are using Corepack to manage your Yarn installation, you can generate an installable archive using `corepack pack -o`. See https://github.com/nodejs/corepack#offline-workflow.

**Reply by Paper-Folding** [reply]
This should be the accepted answer, I did: 
1. Generate pack file on existing project with yarn berry installation via `corepack pack -o yarn.tgz`;  
2. Copy `yarn.tgz` to dev machine;  
In dev machine:  
4. Execute `corepack enable`;  
5. Execute `corepack install -g --cache-only <path/to/yarn.tgz>`;  
6. Set private registry for corepack: `set COREPACK_NPM_REGISTRY=https://foo.com/repository_npm`;  
7. Execute `yarn set version berry` for my project with `"packageManager": "yarn@4.3.1"` already on `package.json`;  
8. Also set private registry for yarn via project's `.yarnrc.yml` file;  
9. Run `yarn install` do the trick.
